mlhandler/README.md

```diff
@@ -0,0 +1,505 @@
+             'max_iter': 100,
+             'multi_class': 'auto',
+             'n_jobs': None,
+             'penalty': 'l2',
+             'random_state': None,
+             'solver': 'lbfgs',
+             'tol': 0.0001,
+             'verbose': 0,
+             'warm_start': False
+         },
+         "model": "LogisticRegression"
+    }
+    </code></pre>
+  </div>
+  <div class="tab-pane fade" id="ajaxcode-params" role="tabpanel" aria-labelledby="ajax-params-tab">
+    <pre><code>
+    $.getJSON('mlhandler?_model')
+    </code></pre>
+  </div>
+  <div class="tab-pane fade" id="curlcode-params" role="tabpanel" aria-labelledby="curl-params-tab">
+    <pre><code>
+    curl -X GET -d @titanic.csv 'mlhandler?_model'
+    </code></pre>
+  </div>
+</div>
+
+
+## Change the model and modify its parameters
+
+An existing model can be replaced with a new one, and all its parameters can be
+modified with a PUT request. The following request replaces the logistic
+regression earlier with a random forest classifier:
+
+```bash
+PUT /mlhandler?_model&class=RandomForestClassififer
+```
+
+Note that at this stage, the model has simply been replaced, but _not_
+retrained. To train it, we can POST to it with `?_action=retrain` parameter as
+mentoined above.
+
+Similarly, any parameter of the model can be changed. For example, to change the
+default number of estimators used in a random forest classifier (100), use:
+
+```bash
+PUT /mlhandler?_model&n_estimators=10
+```
+
+In general, the model's class and any of its parameters can be chained together
+in the PUT request. For example, to change the model to an SGDClassifier with a
+log loss, use:
+
+```bash
+PUT /mlhandler?_model&class=SGDClassifier&loss=log
+```
+
+Any query parameter except `class` which matches the signature of the
+[SGDClassifier](https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.SGDClassifier.html) will be coerced into the model.
+
+Similarly, any of the data transformation options (`include`, `exclude`,
+`cats`, `nums`, etc) can be added or changed at any time with a PUT, as follows:
+
+```bash
+# Ignore PassengerId and Name, consider Embarked as a categorical feature.
+PUT /mlhandler?_model&exclude=PassengerId&exclude=Name&cats=Embarked
+```
+
+
+## Delete a model
+To remove the serialized model from the disk and disable further operations, use
+a delete request as follows:
+
+```bash
+DELETE /mlhandler?_model
+```
+
+# MLHandler Templates
+
+MLHandler supports Tornado templates. You can specify a `template` kwarg as follows:
+
+```yaml
+mlhandler/tutorial:
+  pattern: /$YAMLURL/ml
+  handler: MLHandler
+  template: $YAMLPATH/template.html
+```
+
+The template so specified will be rendered at the `pattern` URL, and it will
+have access to the following variables:
+
+* `{{ handler }}`: The MLHandler instance
+* `{{ handler.model }}`: The sklearn object / estimator / model
+* `{{ data }}`: The training dataset, available as a pandas DataFrame.
```

Comment:
Mention .__class__.__name__ and other estimator attributes included in the template. Mention that these are sklearn objects, but call out the important attributes of those.